src/backend/app/main.py

```diff
@@ -55,12 +58,45 @@ async def save_annotation(request: Request, db: Session = Depends(get_db)):
     db.add(annotation_model)
     db.commit()
 
-    hex_indexes = body["annotation_hexagon_ids"]
-    for hex_index in hex_indexes:
-        hexagon_model = models.AnnotationHexagon()
-        hexagon_model.annotation_id = annotation_model.annotation_id
-        hexagon_model.hex_id = hex_index
```

Comment:
change hex_id to hex_index to be consistent with the database design. This will need to be changed in the models.py file as well!